assets/app/scripts/services/buildConfigs.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("BuildConfigsService", function(){
```

Comment:
I'm definitely a fan of pushing a lot more of our stuff into services than we do, but I can live with keeping these particular functions in the controller if we will not use them anywhere else, per @spadgett  (Side note, BuildConfigService can just `return { foo: func, bar: func }` and skip the prototype stuff).  

Reducing the large functions down to smaller functions with a single responsibility is really the big thing I linger on.